FogCreek/Glitch-Community
# Superb guide: SSR all pages, move seo into react code

## Links
https://superb-guide.glitch.me/
https://github.com/FogCreek/Glitch-Community-Work/issues/214

## Changes:
- Helmet data is used when rendering on the server, so seo fields are set by the page
- All pages render ssr now, which changes /questions, /search, and category pages (they don't data load so they just appear as loaders)
- Created some functions for standalone markdown rendering in utils/,  used for seo descriptions
- Pages had a canonical url of glitch.com/ if it wasn't otherwise specified. I changed it to only include the url tag if it is defined, so that we don't provide bad info

## How To Test:
- Try doing a full load on each page and make sure the seo content is the same as on glitch.com
- Make sure seo images are png and not the svgs used on the page

## Feedback I'm looking for:
- Should anything special be done for search/categories, or is it ok to leave them as loaders?

## Things left to do before deploying:
Merge nova-chance, this pr is based on that branch so it includes all of the changes from it
